File: cs/ch08/Money.Tests/FrancTest.cs
namespace TheSoftwareGorilla.TDD.Money.Tests;

public class FrancTests
{
    
    [SetUp]
    public void Setup()
    {
        // No setup required for these tests
    }

    [Test]
    public void TestConstruction()
    {
        Money five = Money.Franc(5);
        Assert.That(five, Is.Not.Null);
        Assert.That(five, Is.InstanceOf<Franc>());
        Assert.That(five, Is.EqualTo(Money.Franc(5)));
    }

    [Test]
    public void TestMultiplication()
    {
        Money five = Money.Franc(5);
        Assert.That(Money.Franc(10), Is.EqualTo(five.Times(2)));
        Assert.That(Money.Franc(15), Is.EqualTo(five.Times(3)));
    }

}

File: java/ch05/src/test/java/com/thesoftwaregorilla/tdd/money/FrancTest.java
package com.thesoftwaregorilla.tdd.money;

import org.junit.jupiter.api.Test;
import static org.junit.jupiter.api.Assertions.*;


public class FrancTest {

    @Test
    public void testConstruction() {
        Franc five = new Franc(5);
        assertNotNull(five);
        assertEquals(5, five.getAmount());
        Franc ten = new Franc(10);
        assertNotNull(ten);
        assertEquals(10, ten.getAmount());
    }

    @Test
    public void testMultiplication() {
        Franc five = new Franc(5);
        assertEquals(new Franc(10), five.times(2));
        assertEquals(new Franc(15), five.times(3));
    }

    @Test
    public void testEquality() {
        Franc five = new Franc(5);
        assertEquals(five, new Franc(5));
        assertNotEquals(five, new Franc(6));
    }

}
